test/e2e/software_upgrade_e2e_v3_test.go

```diff
@@ -0,0 +1,228 @@
+package e2e
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/suite"
+
+	appparams "github.com/babylonlabs-io/babylon/v3/app/params"
+	v3 "github.com/babylonlabs-io/babylon/v3/app/upgrades/v3"
+	"github.com/babylonlabs-io/babylon/v3/testutil/datagen"
+	"github.com/babylonlabs-io/babylon/v3/testutil/sample"
+	btclighttypes "github.com/babylonlabs-io/babylon/v3/x/btclightclient/types"
+	bstypes "github.com/babylonlabs-io/babylon/v3/x/btcstaking/types"
+
+	"github.com/babylonlabs-io/babylon/v3/app/signingcontext"
+	"github.com/babylonlabs-io/babylon/v3/test/e2e/configurer"
+	"github.com/babylonlabs-io/babylon/v3/test/e2e/configurer/chain"
+	"github.com/babylonlabs-io/babylon/v3/test/e2e/configurer/config"
+)
+
+const (
+	btcstkconsumerModulePath = "btcstkconsumer"
+	zoneconciergeModulePath  = "zoneconcierge"
+)
+
+type SoftwareUpgradeV3TestSuite struct {
+	suite.Suite
+
+	configurer            *configurer.UpgradeConfigurer
+	balancesBeforeUpgrade map[string]sdk.Coin
+	fp1Addr               string
+	fp2Addr               string
+	r                     *rand.Rand
+	fp1BTCSK              *btcec.PrivateKey
+	fp2BTCSK              *btcec.PrivateKey
+	fp1                   *bstypes.FinalityProvider
+	fp2                   *bstypes.FinalityProvider
+}
+
+func (s *SoftwareUpgradeV3TestSuite) SetupSuite() {
+	s.T().Skip("Temporarily skipping v3 upgrade test")
+	s.T().Log("setting up e2e integration test suite for v2.2.0 to v3 upgrade...")
+	var err error
+	s.balancesBeforeUpgrade = make(map[string]sdk.Coin)
+
+	btcHeaderGenesis := sample.SignetBtcHeader195552(s.T())
+
+	// func runs right before the upgrade proposal is sent
+	preUpgradeFunc := func(chains []*chain.Config) {
```

Comment:
all pre upgrade state should happen in this `preUpgradeFunc` i.e if we want to check whther finality providers were migrated correctly: 
- we should create them here in this callback 
- later in `TestUpgradeV3`  check that they have valid chain-id

Basically, all things that happen in `TestUpgradeV3` test, are already after upgrade happens